src/kaleidoscope/device/dygma/Raise.cpp

```diff
@@ -0,0 +1,599 @@
+void twi_disable(void) {
+  Wire.end();
+}
+void twi_init(uint16_t clock_khz) {
+  Wire.begin();
+  Wire.setClock(clock_khz * 1000);
+}
+
+namespace kaleidoscope {
+namespace device {
+namespace dygma {
+
+/********* RaiseHands *********/
+
+struct RaiseHands {
+  static raise::Hand leftHand;
+  static raise::Hand rightHand;
+
+  static void setup();
+  static void initializeSides();
+
+  static uint8_t ansi_iso;
+
+  static void setSidePower(bool power);
+  static bool getSidePower() {
+    return side_power_;
+  }
+
+  static void keyscanInterval(uint16_t interval);
+  static uint16_t keyscanInterval() {
+    return keyscan_interval_;
+  }
+
+ private:
+  static uint16_t keyscan_interval_;
+  static bool side_power_;
+  static uint16_t settings_base_;
+};
+
+raise::Hand RaiseHands::leftHand(0);
+raise::Hand RaiseHands::rightHand(1);
+uint8_t RaiseHands::ansi_iso;
+bool RaiseHands::side_power_;
+uint16_t RaiseHands::settings_base_;
+uint16_t RaiseHands::keyscan_interval_ = 50;
+
+void RaiseHands::setSidePower(bool power) {
+  digitalWrite(SIDE_POWER, power ? HIGH : LOW);
+  side_power_ = power;
+}
+
+void RaiseHands::setup() {
+  settings_base_ = ::EEPROMSettings.requestSlice(sizeof(keyscan_interval_));
+
+  // If keyscan is max, assume that EEPROM is uninitialized, and store the
+  // defaults.
+  uint16_t interval;
+  Kaleidoscope.storage().get(settings_base_, interval);
+  if (interval == 0xffff) {
+    Kaleidoscope.storage().put(settings_base_, keyscan_interval_);
+    Kaleidoscope.storage().commit();
+  }
+  Kaleidoscope.storage().get(settings_base_, keyscan_interval_);
+}
+
+void RaiseHands::keyscanInterval(uint16_t interval) {
+  leftHand.setKeyscanInterval(interval);
+  rightHand.setKeyscanInterval(interval);
+  keyscan_interval_ = interval;
+  Kaleidoscope.storage().put(settings_base_, keyscan_interval_);
+  Kaleidoscope.storage().commit();
+}
+
+void RaiseHands::initializeSides() {
+  // key scan interval from eeprom
+  leftHand.setKeyscanInterval(keyscan_interval_);
+  rightHand.setKeyscanInterval(keyscan_interval_);
+
+  // get ANSI/ISO at every side replug
+  uint8_t l_ansi_iso = leftHand.readANSI_ISO();
+  uint8_t r_ansi_iso = rightHand.readANSI_ISO();
+
+  // setup ansi_iso variable, this will affect led mapping - defaults to ISO if nothing reported
+  if (l_ansi_iso == ANSI || r_ansi_iso == ANSI)
+    ansi_iso = ANSI;
+  else
+    ansi_iso = ISO;
+
+  /*
+   * if the neuron starts up with no sides connected, it will assume ISO. This
+   * turns on an extra LED (hardware LED 19 on left side). If an ANSI left is
+   * then plugged in, the keyboard will switch to ANSI, but LED 19 can't get
+   * wiped because the ANSI LED map doesn't include this LED. It will be driven
+   * from the SLED1735's memory with the same colour as before, which causes
+   * weird looking colours to come on on other seemingly unrelated keys. So: on
+   * a replug, set LED 19 to off to be safe.
+   */
+  leftHand.led_data.leds[19] = {0, 0, 0};
```

Comment:
I'd break out the 19 to a named something.

Reply:
I can name it `ISO_ONLY_LED`, I guess, but I feel that the extensive comments above it should be sufficient, given that this is the only place we treat this led in a special way in code.